### PR TITLE
feat: rewrote action as script

### PR DESCRIPTION
**Pull request by bradegler**

### CODE REVIEW

File: .github/actions/mint-token/action.yml [hunk 1]
```diff
@@ -32,16 +35,29 @@ runs:
...
+                      'Content-Type': 'application/json',
+                      'X-GitHub-OIDC-Token': id_token_value,
+                      'Authorization': `Bearer ${TOKEN}`,
+                  }
+              });
+              const resp = await response.json();
+              core.setOutput('token', resp.token);
+          }
+          catch (err) {
+              core.error(`Error requesting token ${err}`);
```
Comment (verbanicm) [inline .github/actions/mint-token/action.yml:62]: `core.setFailed()` will log the error and set a failed exit code

Comment (bradegler) [reply .github/actions/mint-token/action.yml]: Done.

File: .github/actions/mint-token/action.yml [hunk 2]
```diff
@@ -32,16 +35,29 @@ runs:
...
-        echo "token=$(echo $TOKEN)" >> $GITHUB_OUTPUT;
-
+        TOKEN: '${{ steps.auth.outputs.id_token }}'
+      uses: actions/github-script@v6
+      with:
+        script: |
+          try {
+              const {TOKEN} = process.env;
+              const service_url = core.getInput('service_url', { required: true });
+              const request_body = core.getInput('requested_permissions', { required: true});
```
Comment (verbanicm) [inline .github/actions/mint-token/action.yml:46]: nit: js prefer lowerCamelCase

File: .github/actions/mint-token/action.yml [hunk 3]
```diff
@@ -32,16 +35,29 @@ runs:
...
-          ${{ inputs.service_url}}/token);
-        TOKEN=$(echo ${APP_TOKEN} | jq -j '.token');
-        echo "token=$(echo $TOKEN)" >> $GITHUB_OUTPUT;
-
+        TOKEN: '${{ steps.auth.outputs.id_token }}'
+      uses: actions/github-script@v6
+      with:
+        script: |
+          try {
+              const {TOKEN} = process.env;
```
Comment (shan4993) [inline .github/actions/mint-token/action.yml:44]: nit: do we care about spacing when pulling out variables? i.e. `const { Token } = process.env`

If not feel free to ignore.

File: .github/actions/mint-token/action.yml [hunk 4]
```diff
@@ -32,16 +35,29 @@ runs:
...
+                  method: 'post',
+                  body: request_body,
+                  headers: {
+                      'Content-Type': 'application/json',
+                      'X-GitHub-OIDC-Token': id_token_value,
+                      'Authorization': `Bearer ${TOKEN}`,
+                  }
+              });
+              const resp = await response.json();
+              core.setOutput('token', resp.token);
```
Comment (shan4993) [inline .github/actions/mint-token/action.yml:59]: If this call fails, will we get a NPE on `resp` or is it guaranteed to have a response? If the former, we should check the resp variable for falsey. If the latter, ignore me.

Comment (verbanicm) [reply .github/actions/mint-token/action.yml:59]: I believe for fetch, if an HTTP error code is received, it will fall to the catch block.

**Review by verbanicm — APPROVED** [review]

**Review by sethvargo — APPROVED** [review]

**Pull request merged**